fdbserver/DataDistribution.actor.cpp

```diff
@@ -1300,14 +1312,14 @@ struct DDTeamCollection : ReferenceCounted<DDTeamCollection> {
 		return maxMatchingServers;
 	}
 
-	int overlappingMachineMembers(vector<Standalone<StringRef>>& team) {
+	int overlappingMachineMembers(vector<Standalone<StringRef>> const& team) const {
 		if (team.empty()) {
 			return 0;
 		}
 
 		int maxMatchingServers = 0;
-		Standalone<StringRef>& serverID = team[0];
-		for (auto& usedTeam : machine_info[serverID]->machineTeams) {
+		auto const& serverID = team[0];
+		for (auto const& usedTeam : machine_info.at(serverID)->machineTeams) {
```

Comment:
How confident are we that `serverID` is always a key in `machine_info` ? Should we consider using `find` and asserting it's not `end()` here? That way we don't get difficult to debug `unknown_error`'s

Reply:
Previously if `serverID` was not in `machine_info`, this would have segfaulted because we immediately attempt to dereference `machine_info[serverID]`. So the `unknown_error` that would now result is nicer than what would have happened before.

Reply:
IMO segfaulting would be nicer since we'd get exactly the right stack trace we'd need to identify the problem

Reply:
I'll add an assertion here, so we will get the stack trace through the assertion.